kratos/python_scripts/analysis_stage.py

```diff
@@ -280,6 +361,32 @@ def _GetSimulationName(self):
         """
         return "Analysis"
 
+    def _CheckIfModelIsModified(self):
+        # We check of some model part requires reinitialize
+        modified_model = False
+        if hasattr(self, 'model'):
+            model_part_names = self.model.GetModelPartNames()
+            for model_part_name in model_part_names:
+                if self.model.GetModelPart(model_part_name).Is(KratosMultiphysics.MODIFIED):
+                    modified_model = True
+                    break
+
+        return modified_model
+
+    def _SetModelIsModified(self):
```

Comment:
could this function name be more clear on what it does?

Reply:
Done